### PR TITLE
Move version info to version.py

### PR DESCRIPTION
**Pull request by genebean**
This makes it so the code matches GitHub. Since changes have come in post 1.3.0 I have set the initial version to 1.4.0 (aka, the next release)

### CODE REVIEW

**Comment by genebean** [comment]
@metaMMA would you look this over and see if it works for you?

File: setup.pp [hunk 1]
```diff
@@ -110,7 +110,7 @@
   user    => 'pi',
   path    => '/bin:/usr/bin',
   command => 'python3 /home/pi/PiWeatherRock/scripts/upgrade.py',
-  unless  => 'grep "0.0.14" config.json',
+  unless  => 'grep $(cat version.py|cut -d "\'" -f2) config.json',
```
Comment (metaMMA) [inline setup.pp:113]: Should this:
```
unless  => 'grep $(cat version.py|cut -d "\'" -f2) config.json',
```
be
```
unless  => 'grep $(cat version.py|cut -d "\'" -f2)',
```

**Review by metaMMA — COMMENTED** [review]
https://github.com/genebean/PiWeatherRock/blob/66e4cb3020fdc8a4104b7e1422b6d239985b626b/scripts/upgrade.py#L14-L30

I'll need to make a couple changes to these lines, too. Shouldn't be too difficult. PR with amendments found here: https://github.com/genebean/PiWeatherRock/pull/56

**Comment by metaMMA** [comment]
> I'll need to make a couple changes to these lines, too. Shouldn't be too difficult.

```
import json
import os
import socket

pi_ip = socket.gethostbyname(socket.gethostname() + ".local")

if os.path.exists("config.py"):
    print(f"\nImporting current configuration settings.\n\n"
          f"Go to http://{pi_ip}:8888 to view new configuration interface.\n")
    import config
    old_config = {}
    old_config["ds_api_key"] = config.DS_API_KEY
    old_config["update_freq"] = int(config.DS_CHECK_INTERVAL)
    old_config["lat"] = float(config.LAT)
    old_config["lon"] = float(config.LON)
    old_config["units"] = config.UNITS
    old_config["lang"] = config.LANG
    old_config["fullscreen"] = config.FULLSCREEN
    old_config["icon_offset"] = float(config.LARGE_ICON_OFFSET)
    old_config["plugins"] = {}
    old_config["plugins"]["daily"] = {}
    old_config["plugins"]["hourly"] = {}
    old_config["plugins"]["daily"]["enabled"] = True
    old_config["plugins"]["hourly"]["enabled"] = True
    if hasattr(config, "DAILY_PAUSE"):
        old_config["plugins"]["daily"]["pause"] = int(config.DAILY_PAUSE)
    else:
        old_config["plugins"]["daily"]["pause"] = 60
    if hasattr(config, "HOURLY_PAUSE"):
        old_config["plugins"]["hourly"]["pause"] = int(config.HOURLY_PAUSE)
    else:
        old_config["plugins"]["hourly"]["pause"] = 60
    if hasattr(config, "INFO_PAUSE"):
        old_config["info_pause"] = int(config.INFO_PAUSE)
    else:
        old_config["info_pause"] = 300
    if hasattr(config, "INFO_DELAY"):
        old_config["info_delay"] = int(config.INFO_DELAY)
    else:
        old_config["info_delay"] = 900
    os.remove("config.py")
elif os.path.exists("config.json"):
    with open("config.json", "r") as f:
        old_config = json.load(f)
else:
    with open("config.json-sample", "r") as f:
        old_config = json.load(f)
    print(f"\nYou must configure PiWeatherRock.\n\n"
          f"Go to http://{pi_ip}:8888 to configure.\n")

with open("config.json-sample", "r") as f:
    new_config = json.load(f)

# Add any new config variables
for key in new_config.keys():
    if key not in old_config.keys():
        old_config[key] = new_config[key]

with open("config.json", "w") as f:
    json.dump(old_config, f)
```

I think this upgrade script should work for all upgrades moving (If we ever need to remove any config variables from a newer version, then a few more lines would need to be added. But for now, this simply copies over all older config values, and adds any new variable with it's default value.)

**Comment by genebean** [comment]
@metaMMA Look good to you now?

File: setup.pp (hunk 1 above)
Comment (metaMMA) [inline setup.pp:113]: I'm getting the following error when I test line 113 as a standalone bash script (on a full Debian 10 system) :
```
cut: the delimiter must be a single character
```
I've had success with this, though: `grep $(cat version.py|cut -d"'" -f2) config.json`

**Comment by metaMMA** [comment]
> @metaMMA Look good to you now?

Other than the `cut` delimiter error, everything else looks good to me.

**Comment by genebean** [comment]
The `\` is an escape character so the `’` can be in the middle of the puppet parameter.

**Pull request merged**